lib/signaling/v2/peerconnectionmanager.js

```diff
@@ -323,16 +323,17 @@ function applyMediaStreamTrackAndDataTrackSenderChanges(peerConnectionManager, c
     || changes.data.remove.size
     || changes.media.add.size
     || changes.media.remove.size) {
+    peerConnectionManager._peerConnections.forEach(function(peerConnection) {
```

Comment:
@markandrus I've separated out the removal of tracks from their addition that happens later because we need to remove the tracks in the local stream collection before changes are applied to it.

Reply:
> we need to remove the tracks in the local stream collection before changes are applied to it.

Why is that? Also, this suggests to me we ought to move to a MediaStreamTrack- rather than MediaStream-based API soon...

Reply:
@markandrus I agree, however, we do need a MediaStream to pass to `addTrack()`.

Reply:
Yeah, probably we can move the MediaStream yet another level down in the future (and let each PeerConnectionV2 have it's own MediaStream instance; or perhaps pass the same MediaStream to each PeerConnectionV2's constructor). `addTrack` is supposed to let you omit the MediaStream(s) argument completely, but I saw that was unsupported in at least one of the browsers so far.

Reply:
@markandrus I wouldn't create a new MediaStream for PeerConnectionV2 because I think we just fixed a bug where too many MediaStreams was causing audio problems in FF.